tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -138,35 +181,114 @@ declare var cfg:ISnippetConfig;
-            tags[strAiOperationName] = locn && locn.pathname || "_unknown_";
-            tags[strAiSdkVersion] = "javascript:snippet_" + (appInsights.sv || appInsights.version);
-
-            let envelope:IEnvelope = {
-                time: _getTime(),
-                iKey: iKey,
-                name: "Microsoft.ApplicationInsights." + iKey.replace(/-/g, strEmpty) + "." + theType,
-                sampleRate: 100,
-                tags: tags,
-                data: {
-                    baseData: {
-                        ver: 2
-                    }
-                },
-                ver: undefined,
-                seq: "1",
-                aiDataContract: undefined
+        function _addTimeZone(envelope: any): void {
+            // Add time zone
+            const timeZone: number = new Date().getTimezoneOffset();
+            let minutes: number = timeZone % 60;
+            let hours: number = (timeZone - minutes) / 60;
+            let timeZonePrefix: string = "+";
+            if (hours > 0) {
+                timeZonePrefix = "-";
+            }
+            hours = Math.abs(hours);
+            minutes = Math.abs(minutes);
+        
+            envelope.ext = envelope.ext || {};
+            envelope.ext.loc = {  // Add time zone
+                tz: timeZonePrefix + (hours < 10 ? "0" + hours : hours.toString()) + ":" + (minutes < 10 ? "0" + minutes : minutes.toString())
             };
+        }
 
-            return envelope;
+        function _getTenantId(apiKey: string): string {
+            let result: string = "";
+        
+            if (apiKey) {
+                const indexTenantId: number = apiKey.indexOf("-");
+                if (indexTenantId > -1) {
+                    result = apiKey.substring(0, indexTenantId);
+                }
+            }
+            return result;
+        }
+        
+        function _addUser(envelope: any): void {
+            const strUndefined: string = "undefined";
+            // Add user language
+            if (typeof navigator !== strUndefined) {
+                const nav: Navigator & { userLanguage?: string } = navigator;
+                envelope.ext = envelope.ext || {};
+                envelope.ext.user = {
+                    locale: nav.userLanguage || nav.language
+                };
+            }
+        }
+
+        function _createEnvelope(iKey:string, theType:string) {
+            if (_epoch === 0) {
+                _epoch = Math.floor((UInt32Mask * Math.random()) | 0) >>> 0;
+            }
+            if (isOneDS){
+                let envelope: oneDsEnvelope = {
+                    data: {
+                        baseData: {
+                            ver: 2
+                        }
+                    },
+                    ext: {
+                        app: { sesId: "0000" },
+                        intweb: {},
+                        sdk: {
+                            ver: "javascript:" + OneDSstrSnippetVersion,
```

Comment:
As we are creating a new "unified" (ish) snippet lets also make the reported version the same for both AI and 1DS